frontend/src/views/admin/assignments/confirm-with-dialog-action-button.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { ActionButton } from "../../../components/action-buttons";
+import { OfferConfirmationDialog } from "./offer-confirmation-dialog";
+import { Assignment } from "../../../api/defs/types";
+
+export function ConfirmWithDialogActionButton(props: {
+    buttonIcon: Object;
+    buttonEnable: boolean;
+    actionConfirmation: () => void;
+    buttonText: string;
+    selectedAssignments: Assignment[];
+    dialogVisible: boolean;
+    dialogSetVisible: Function;
+    actionCallback: Function;
+    dialogTitle: string;
+    dialogBody: string;
+    dialogConfirmation: string;
+}) {
+    const {
+        buttonIcon,
+        buttonEnable,
+        actionConfirmation,
+        buttonText,
+        selectedAssignments,
+        dialogVisible,
+        dialogSetVisible,
+        actionCallback,
+        dialogTitle,
+        dialogBody,
+        dialogConfirmation,
+    } = props;
```

Comment:
You can move this right into the function definition. I.e., use destructuring there instead of temporarily assigning to `props`.